backend/src/main/java/gov/cdc/usds/simplereport/api/converter/FhirConverter.java

```diff
@@ -273,4 +281,66 @@ public static Extension convertToTribalAffiliationExtension(String tribalAffilia
     }
     return null;
   }
+
+  public static Device convertToDevice(DeviceType deviceType) {
+    if (deviceType != null) {
+      return convertToDevice(
+          deviceType.getManufacturer(),
+          deviceType.getModel(),
+          Objects.toString(deviceType.getInternalId(), ""));
+    }
+    return null;
+  }
+
+  public static Device convertToDevice(String manufacturer, String model, String id) {
+    if (StringUtils.isNotBlank(manufacturer) || StringUtils.isNotBlank(model)) {
+      var device =
+          new Device()
+              .setManufacturer(manufacturer)
+              .addDeviceName(
+                  new DeviceDeviceNameComponent().setName(model).setType(DeviceNameType.MODELNAME));
+      device.setId(id);
+      return device;
+    }
+    return null;
+  }
+
+  public static Specimen convertToSpecimen(
+      String specimenCode,
+      String specimenName,
+      String collectionCode,
+      String collectionName,
+      String id) {
+    var specimen = new Specimen();
+    specimen.setId(id);
+    if (StringUtils.isNotBlank(specimenCode)) {
+      var codeableConcept = specimen.getType();
+      var coding = codeableConcept.addCoding();
+      coding.setSystem(SNOMED_CODE_SYSTEM);
+      coding.setCode(specimenCode);
+      codeableConcept.setText(specimenName);
+    }
+    if (StringUtils.isNotBlank(collectionCode)) {
+      var collection = specimen.getCollection();
+      var codeableConcept = collection.getBodySite();
+      var coding = codeableConcept.addCoding();
+      coding.setSystem(SNOMED_CODE_SYSTEM);
+      coding.setCode(collectionCode);
+      codeableConcept.setText(collectionName);
+    }
+
+    return specimen;
+  }
+
+  public static Specimen convertToSpecimen(SpecimenType specimenType) {
+    if (specimenType != null) {
```

Comment:
Should we even allow null SpecimenTypes to be passed in here? I'm wondering if it's worth just making the parameter `@NotNull`.

Reply:
Around this specifically, I think there will be some inconsistency that needs to be fixed when all of these get merged in.